flixel/system/frontEnds/CollisionFrontEnd.hx

```diff
@@ -0,0 +1,475 @@
+package flixel.system.frontEnds;
+
+import flixel.FlxG;
+import flixel.FlxObject;
+import flixel.math.FlxPoint;
+import flixel.math.FlxRect;
+import flixel.tile.FlxBaseTilemap;
+import flixel.util.FlxDirectionFlags;
+
+using flixel.util.FlxCollision;
+
+@:access(flixel.FlxObject)
+class CollisionFrontEnd
+{
+	public function new () {}
+	
+	/**
+	 * This value dictates the maximum number of pixels two objects have to intersect
+	 * before collision stops trying to separate them.
+	 * Don't modify this unless your objects are passing through each other.
+	 */
+	public var maxOverlap:Float = 4;
+	
+	/**
+	 * Call this function to see if one `FlxObject` overlaps another within `FlxG.worldBounds`.
+	 * Can be called with one object and one group, or two groups, or two objects,
+	 * whatever floats your boat! For maximum performance try bundling a lot of objects
+	 * together using a `FlxGroup` (or even bundling groups together!).
+	 *
+	 * NOTE: does NOT take objects' `scrollFactor` into account, all overlaps are checked in world space.
+	 *
+	 * NOTE: this takes the entire area of `FlxTilemap`s into account (including "empty" tiles).
+	 * Use `FlxTilemap#overlaps()` if you don't want that.
+	 *
+	 * @param   a        The first object or group you want to check.
+	 * @param   b        The second object or group you want to check. Can be the same group as the first.
+	 * @param   notify   Called on every object that overlaps another.
+	 * @param   process  Called on every object that overlaps another, determines whether to call the `notify`.
+	 * @return  Whether any overlaps were detected.
+	 */
+	overload public inline extern function overlap<TA:FlxObject, TB:FlxObject>(a, ?b, ?notify:(TA, TB)->Void, ?process:(TA, TB)->Bool)
```

Comment:
I like this change as I almost always write the first two lines of my notify/process func as
```
var player:Player = cast a;
var other:Enemy = cast e;
```

Will this blow up if someone collides/overlaps a group that has the wrong types in it? More concretely if they do something like `overlap<Player, Coin>(player, allCollidableObjectsInGame, (p, c) -> { ... });`

I am not in a spot where I can just run the code, but wanted to ask while it's on my mind so I don't lose the thought.

Reply:
I still need some testing, but i dont think it will any worse than using wrong types with the current system, which just uses Dynamics.  One benefit of this is that it ensures the notify and process use the same types